atlas/portal/pkg/svc/zserver.go

```diff
@@ -0,0 +1,219 @@
+	emptyRequest       = "empty Request"
+	success            = "success"
+	errorMsg           = "error"
+	errorMissingResp   = "missing repository response error"
+	hiddenUptimeMsg    = "uptime is hidden, mode = false"
+	portal             = "portal"
+	responder          = "responder"
+	storage            = "storage"
+)
+
+type Portal struct {
+	pb.PortalServer
+	client responderpb.ResponderClient
+	logger *logrus.Logger
+	portal models.Service
+}
+
+func getPortal() models.Service {
+	return models.Service{
+		ServiceName:          "portal",
+		ServiceDesc:          "portal service desc",
+		ServiceUptime:        time.Now(),
+		ServiceCountRequests: 0,
+	}
+}
+
+func NewPortal(logger *logrus.Logger) (*Portal, error) {
+	conn, err := grpc.Dial("0.0.0.0:9091" /*+os.Getenv("PORT")*/, grpc.WithInsecure(), grpc.WithBlock())
+	if err != nil {
+		logger.Fatalf("\nDid not connect %v\n", err)
+	}
+
+	client := responderpb.NewResponderClient(conn)
+
+	return &Portal{
+		client: client,
+		logger: logger,
+		portal: getPortal(),
+	}, nil
+}
+
+func (p *Portal) GetVersion(context.Context, *empty.Empty) (*pb.VersionResponse, error) {
+	return &pb.VersionResponse{Version: version}, nil
+}
+
+func (p *Portal) GetInfo(ctx context.Context, in *pb.GetInfoRequest) (*pb.GetInfoResponse, error) {
+	if in == nil || in.Service == "" {
+		return nil, errors.New(emptyRequest)
+	}
+
+	if in.Service == portal {
+		p.portal.ServiceCountRequests++
+		return &pb.GetInfoResponse{Value: p.portal.ServiceDesc}, nil
+	}
+
+	if in.Service == responder || in.Service == storage {
+		resp, err := p.client.GetInfo(ctx, &responderpb.GetInfoRequest{Service: in.Service})
+		if err != nil {
+			return nil, err
+		}
+		return &pb.GetInfoResponse{Value: resp.Value}, nil
+	}
+
+	return nil, errors.New(invalidServiceName)
+}
+
+func (p *Portal) SetInfo(ctx context.Context, in *pb.SetInfoRequest) (*pb.SetInfoResponse, error) {
+	if in == nil || in.Service == "" || in.Value == "" {
+		return nil, errors.New(emptyRequest)
+	}
+
+	if in.Service == portal {
+		p.portal.ServiceCountRequests++
+		p.portal.ServiceDesc = in.Value
+		return &pb.SetInfoResponse{Msg: success}, nil
+	}
+
+	if in.Service == responder || in.Service == storage {
+		resp, err := p.client.SetInfo(ctx, &responderpb.SetInfoRequest{Service: in.Service, Value: in.Value})
+		if err != nil {
+			return nil, err
+		}
+		return &pb.SetInfoResponse{Msg: resp.Msg}, nil
+	}
+
+	return nil, errors.New(invalidServiceName)
+}
+
+func (p *Portal) GetUptime(ctx context.Context, in *pb.GetUptimeRequest) (*pb.GetUptimeResponse, error) {
+	if in == nil || in.Service == "" {
+		return nil, errors.New(emptyRequest)
+	}
+
+	if in.Service == portal {
+		p.portal.ServiceCountRequests++
+	}
+
+	if in.Service == portal || in.Service == responder || in.Service == storage {
+		resp, err := p.client.GetUptime(ctx, &responderpb.GetUptimeRequest{Service: in.Service})
+		if err != nil {
+			return nil, err
+		}
+
+		if in.Service == portal {
```

Comment:
User can check the service uptime. Responder will answer only when its mode is "true", or user should get something like "service is temporarily disabled". 

I think that is only for Responder, not for Portal.